dev/tools/coqdev.el

```diff
@@ -0,0 +1,102 @@
+;;; coqdev.el --- Emacs helpers for Coq development  -*- lexical-binding:t -*-
+
+;; Copyright (C) 2018 The Coq Development Team
+
+;; Maintainer: coqdev@inria.fr
+
+;;; Commentary:
+
+;; Helpers to set compilation commands, proof general variables, etc
+;; for Coq development
+
+;; You can disable individual features without editing this file by
+;; using `remove-hook', for instance
+;; (remove-hook 'hack-local-variables-hook #'coqdev-setup-compile-command)
+
+;;; Installation:
+
+;; To use this, with coqdev.el located at /path/to/coqdev.el, add the
+;; following to your init:
+
+;; (add-to-list 'load-path "/path/to/coqdev/")
+;; (require 'coqdev)
+
+;; If you load this file from a git repository, checking out an old
+;; commit will make it disappear and cause errors for your Emacs
+;; startup. To ignore those errors use (require 'coqdev nil t). If you
+;; check out a malicious commit Emacs startup would allow it to run
+;; arbitrary code, to avoid this you can copy coqdev.el to any
+;; location and adjust the load path accordingly (of course if you run
+;; ./configure to compile Coq it is already too late).
+
+;;; Code:
+
+(require 'subr-x)
+
+(defun coqdev-default-directory ()
+  "Return the Coq repository containing `default-directory'."
+  (when-let ((dir (locate-dominating-file default-directory "META.coq")))
+    (expand-file-name dir)))
+
+(defun coqdev-setup-compile-command ()
+  "Setup `compilate-command' for Coq development."
+  (when-let ((dir (coqdev-default-directory)))
+      (setq-local compile-command (concat "make -C " dir))))
+(add-hook 'hack-local-variables-hook #'coqdev-setup-compile-command)
+
+(defvar camldebug-command-name) ; from camldebug.el (caml package)
+(defvar ocamldebug-command-name) ; from ocamldebug.el (tuareg package)
+(defun coqdev-setup-camldebug ()
+  "Setup ocamldebug for Coq development.
+
+Specifically `camldebug-command-name' and `ocamldebug-command-name'."
+  (when-let ((dir (coqdev-default-directory)))
+    (setq-local camldebug-command-name (concat dir "dev/ocamldebug-coq"))
+    (setq-local ocamldebug-command-name (concat dir "dev/ocamldebug-coq"))))
```

Comment:
`expand-file-name` is usually better than `concat`

Reply:
concat is shorter and we already expanded in coqdev-default-directory, is there still some use we can get out of expand-file-name?

Reply:
Not too much. I don't think expand-file-name is guaranteed to return a path ending with a "/", but in practice it likely does in most cases.